### PR TITLE
[Keyboard]Add X-Bows Knight_Plus Keyboard

### PR DESCRIPTION
**Pull request by XBowsTech**
## Description

Add support for X-Bows Knight_Plus Keyboard and Numpad Keyboard.
This is Knight_Plus keyboard set that contains Knight_Plus and Numpad.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/xbows/knight_plus/config.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+#define DIODE_DIRECTION COL2ROW
+
+#define DEBOUNCE 3
+
+#ifdef RGB_MATRIX_ENABLE
+#    define RGB_MATRIX_LED_PROCESS_LIMIT 18
+#    define RGB_MATRIX_LED_FLUSH_LIMIT 16
+#    define RGB_DISABLE_AFTER_TIMEOUT 0          // number of ticks to wait until disabling effects
+#    define RGB_MATRIX_KEYPRESSES
+#    define RGB_DISABLE_WHEN_USB_SUSPENDED true  // turn off effects when suspended
```
Comment (drashna) [inline keyboards/xbows/knight_plus/config.h:39]:
```suggestion
#    define RGB_DISABLE_WHEN_USB_SUSPENDED  // turn off effects when suspended
```

File: keyboards/xbows/knight_plus/knight_plus.c [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+void suspend_power_down_kb(void) {
+    rgb_matrix_set_suspend_state(true);
+    suspend_power_down_user();
+}
+
+void suspend_wakeup_init_kb(void) {
+    rgb_matrix_set_suspend_state(false);
+    suspend_wakeup_init_user();
+}
```
Comment (drashna) [inline keyboards/xbows/knight_plus/knight_plus.c:146]:
```suggestion
```

File: keyboards/xbows/numpad/config.h [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+#define MATRIX_COL_PINS { D4, D6, B2, B1 }
+#define DIODE_DIRECTION COL2ROW
+
+#define DEBOUNCE 3
+
+#ifdef RGB_MATRIX_ENABLE
+#    define RGB_MATRIX_LED_PROCESS_LIMIT 18
+#    define RGB_MATRIX_LED_FLUSH_LIMIT 16
+#    define RGB_DISABLE_AFTER_TIMEOUT 0          // number of ticks to wait until disabling effects
+#    define RGB_DISABLE_WHEN_USB_SUSPENDED true  // turn off effects when suspended
```
Comment (drashna) [inline keyboards/xbows/numpad/config.h:38]:
```suggestion
#    define RGB_DISABLE_WHEN_USB_SUSPENDED  // turn off effects when suspended
```

File: keyboards/xbows/numpad/numpad.c [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+void suspend_power_down_kb(void) {
+    rgb_matrix_set_suspend_state(true);
+    suspend_power_down_user();
+}
+
+void suspend_wakeup_init_kb(void) {
+    rgb_matrix_set_suspend_state(false);
+    suspend_wakeup_init_user();
+}
```
Comment (drashna) [inline keyboards/xbows/numpad/numpad.c:82]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/xbows/knight_plus/knight_plus.c (hunk 2 above)
Comment (XBowsTech) [reply keyboards/xbows/knight_plus/knight_plus.c:146]: Can you tell me why you want to delete these lines of code?

File: keyboards/xbows/numpad/numpad.c (hunk 4 above)
Comment (XBowsTech) [reply keyboards/xbows/numpad/numpad.c:82]: I found a bug. The computer was in sleep state, and the RGB matrix could not be automatically closed. And the light animation gets stuck until the computer wakes up again.

File: keyboards/xbows/knight_plus/config.h (hunk 1 above)
Comment (XBowsTech) [reply keyboards/xbows/knight_plus/config.h]: This correction appears to cause some errors

File: keyboards/xbows/numpad/config.h (hunk 3 above)
Comment (XBowsTech) [reply keyboards/xbows/numpad/config.h]: This correction appears to cause some errors

Comment (XBowsTech) [reply keyboards/xbows/numpad/config.h]: > 
This correction appears to cause some errors

**Pull request closed without merging**